linux-armv6/Dockerfile.in

```diff
@@ -1,48 +1,41 @@
 FROM dockcross/base:latest
 MAINTAINER Matt McCormick "matt.mccormick@kitware.com"
 
-# Enable 32 bits binaries
-RUN dpkg --add-architecture i386 && \
-    apt-get update && \
-    apt-get install -y libstdc++6:i386 libgcc1:i386 zlib1g:i386
+# Crosstool-ng version
+# We use a git commit while waiting for 1.25 to release (1.24 is several years old)
+ENV CT_VERSION 9e10cb6fbaab6d83e1305b2d53ab585d14989667
+
+# This is for 32-bit ARMv6 Linux
+# Raspberry Pi is ARMv6+VFP2
+#include "common.crosstool"
 
 # The cross-compiling emulator
-RUN apt-get update && apt-get install -y \
+RUN apt-get update \
+&& apt-get install -y \
   qemu-user \
-  qemu-user-static
-
-ENV CROSS_TRIPLE arm-linux-gnueabihf
-ENV CROSS_ROOT /usr/${CROSS_TRIPLE}
-ENV AS=/usr/bin/${CROSS_TRIPLE}-as \
-    AR=/usr/bin/${CROSS_TRIPLE}-ar \
-    CC=/usr/bin/${CROSS_TRIPLE}-gcc \
-    CPP=/usr/bin/${CROSS_TRIPLE}-cpp \
-    CXX=/usr/bin/${CROSS_TRIPLE}-g++ \
-    LD=/usr/bin/${CROSS_TRIPLE}-ld \
-    FC=/usr/bin/${CROSS_TRIPLE}-gfortran
-
-# Raspberry Pi is ARMv6+VFP2, Debian armhf is ARMv7+VFP3
-# Since this Dockerfile is targeting linux-arm from Raspberry Pi onward,
-# we're sticking with it's custom built cross-compiler with hardfp support.
-# We could use Debian's armel, but we'd have softfp and loose a good deal
-# of performance.
-# See: https://wiki.debian.org/RaspberryPi
-# We are also using the 4.7 version of the toolchain, so that glibc=2.13
-
-# Instead of cloning the whole repo (>1GB at the of writing this), we want to do a so-called "sparse checkout" with "shallow cloning":
-# https://stackoverflow.com/questions/600079/is-there-any-way-to-clone-a-git-repositorys-sub-directory-only/13738951#13738951
-
-RUN mkdir rpi_tools && cd rpi_tools && git init && git remote add -f origin https://github.com/raspberrypi/tools && \
-    git config core.sparseCheckout true && echo "arm-bcm2708/gcc-linaro-${CROSS_TRIPLE}-raspbian" >> .git/info/sparse-checkout && \
-    git pull --depth=1 origin master && rsync -av arm-bcm2708/gcc-linaro-${CROSS_TRIPLE}-raspbian/ /usr/ && rm -rf ../rpi_tools
```

Comment:
Looking at this, and more specifically:

> Since this Dockerfile is targeting linux-arm from Raspberry Pi onward

Does it still work for RPi with those changes? In the removed code there is something happening with bcm2708, but I am not clear exactly what it does...

Reply:
> Looking at this, and more specifically:
> 
> > Since this Dockerfile is targeting linux-arm from Raspberry Pi onward
> 
> Does it still work for RPi with those changes? In the removed code there is something happening with bcm2708, but I am not clear exactly what it does...

I don't have a raspberry pi 1, only the 3B and 4B, I tested with QEMU and docker,
If you are interested to know how I tested the executable:

Build exec:

`./linux-armv6.sh bash -c '$CC hello.c -o hello_armv6'`
(WIth DEFAULT_DOCKCROSS_IMAGE=dockcross/linux-armv6:20210628-83e35ef for me)

Check exec:
`readelf -a -W hello_armv6`

Upload file to : https://www.file.io/

Run docker container with QEMU and raspbian image : 
`docker run -it --rm lukechilds/dockerpi pi1`
**login**: _pi_
**password**: _raspberry_

Get file and run:
`wget <URL> -O hello_armv6`
`chmod +x hello_armv6`
`./hello_armv6`

It displays: `Hello cross-compilation world!`

More: https://github.com/lukechilds/dockerpi

Reply:
@thewtex @jcfr Do you use this image for RPi? If yes would you mind reviewing?

Reply:
I'll dig up what Pi's I have around and test this weekend.